code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm

```diff
@@ -155,131 +119,115 @@
-					<span class='[get_component_span(i)]_large'>[required_components[i]]</span>")
-		else
+	if(!countdown)
+		if(is_servant_of_ratvar(user))
+			to_chat(user, "Ratvar still hasn't finished preparing. The clock is ticking; get building!")
+	else
+		if(is_servant_of_ratvar(user) || isobserver(user))
 			to_chat(user, "<span class='big'><b>Seconds until Ratvar's arrival:</b> [get_arrival_text(TRUE)]</span>")
 			switch(progress_in_seconds)
 				if(-INFINITY to GATEWAY_REEBE_FOUND)
-					to_chat(user, "<span class='heavy_brass'>It's still opening.</span>")
+					to_chat(user, "<span class='heavy_brass'>The Ark is trying to breach the walls of Ratvar's prison.</span>")
 				if(GATEWAY_REEBE_FOUND to GATEWAY_RATVAR_COMING)
-					to_chat(user, "<span class='heavy_brass'>It's reached the Celestial Derelict and is drawing power from it.</span>")
+					to_chat(user, "<span class='heavy_brass'>The outer walls of the Derelict have fallen. The inner reinforcements are still being melted.</span>")
 				if(GATEWAY_RATVAR_COMING to INFINITY)
-					to_chat(user, "<span class='heavy_brass'>Ratvar is coming through the gateway!</span>")
-	else
+					to_chat(user, "<span class='heavy_brass'>The Derelict has been breached! Ratvar is coming through!</span>")
+		else
+			switch(progress_in_seconds)
+				if(-INFINITY to GATEWAY_REEBE_FOUND)
+					to_chat(user, "<span class='warning'>You can make out a huge wall made of strange runed metal, slowly melting away.</span>")
+				if(GATEWAY_REEBE_FOUND to GATEWAY_RATVAR_COMING)
+					to_chat(user, "<span class='warning'>You can make out a melting, metal wall.</span>")
+				if(GATEWAY_RATVAR_COMING to INFINITY)
+					to_chat(user, "<span class='boldwarning'>A huge shape is slowly coming through the gateway!</span>")
+
+/obj/structure/destructible/clockwork/massive/celestial_gateway/process()
+	wisdom_regen_cycle++
+	if(wisdom_regen_cycle >= ARK_WISDOM_REGEN)
+		adjust_clockwork_wisdom(1)
+		wisdom_regen_cycle = 0
+	if(countdown)
+		if(!first_sound_played || prob(7))
+			for(var/M in GLOB.player_list)
+				var/mob/MO = M
+				if(M && !isnewplayer(M) && MO.z == z)
+					to_chat(M, "<span class='warning'><b>You hear otherworldly sounds from the [dir2text(get_dir(get_turf(M), get_turf(src)))]...</span>")
+		if(!obj_integrity)
+			return 0
+		var/convert_dist = 1 + (round(Floor(progress_in_seconds, 15) * 0.067))
+		for(var/t in RANGE_TURFS(convert_dist, loc))
+			var/turf/T = t
+			if(!T)
+				continue
+			if(get_dist(T, src) < 2)
+				if(iswallturf(T))
+					var/turf/closed/wall/W = T
+					W.dismantle_wall()
+				else if(t && (isclosedturf(T) || !is_blocked_turf(T)))
+					T.ChangeTurf(/turf/open/floor/clockwork)
+			var/dist = cheap_hypotenuse(T.x, T.y, x, y)
+			if(dist < convert_dist)
+				T.ratvar_act(FALSE, TRUE, 3)
+		for(var/obj/O in orange(1, src))
+			if(!O.pulledby && !istype(O, /obj/effect) && O.density)
+				if(!step_away(O, src, 2) || get_dist(O, src) < 2)
```

Comment:
I didn't code it, don't shoot the messenger. It's to destroy stuff like tables etc. that get built around the Ark, I'm pretty sure, which I think is probably a good thing.